Date-fns v2

jariz opened the pull request:
So, this PR contains my effort to get this working with v2.
Sadly I was unable to get this fully working, but it's a pretty good start.

I've fixed the ToC selector and disabled CORS within puppeteer because the site was attempting to fetch something from a different domain (firebase).
I've also increased the timeout a bit because it was grabbing the HTML while the docs were still in a loading state (but that might just be my bad laptop)

The main problem is that dashing seems to get stuck on the first HTML file it can find. I have not been able to find out why.

jariz committed:
WIP to get project working with datefns v2

jariz commented:
@iansinnott could you possibly take a look?

iansinnott commented:
Hey @jariz thanks for the contribution. 

Hm, disabling web security should not be necessary. The site was fetching from firebase when I initially created these docs. That's part of why I spun up a whole headless browser if I remember correctly, rather than simply parsing HTML.

What makes you think CORS was causing an issue?

jariz commented:
> What makes you think CORS was causing an issue?

`utils.js`
```
const browser = await puppeteer.launch({
    headless: false
  });
```

![](https://jari.lol/CvHs3slcFB.png)

😛 

It could all be my environment though.

Perhaps you could give it a try on your machine and see if it works? 
All I want is to be able to search through dateFns from dash again because I kinda can't stand their site lol.

iansinnott commented:
Ah i see. that's because you're loading from localhost. Chrome supports CORS just fine, and thus headless chrome does too. The issue is that the website sees itself as localhost and is trying to hit assets at some other domain that doesn't allow requests from localhost.

You could use the live site rather than localhost

iansinnott commented:
It's been a while since I've touched this code, and I'm not totally sure what my motivation was for doing it all locally. Anyway, if you parse the docs from the live site rather than locally there's also no need for the whole local server that is spawned.

You might be able to just change this line to point to the live docs: https://github.com/iansinnott/date-fns-dash-docset/blob/master/gen-static.js#L93

jariz committed:
Now works fully with dfns 2.0

- Use latest version of dashing because 3.0 was crashing
- Strip navigation from pages to prevent them from showing up in Dash
- Use page headers as the selector instead of nav items.

jariz commented:
Hi @iansinnott, happy new year!
Had some more time to work on this and I finished this PR up.

Regarding what you said here: 
No idea why you chose to serve it locally, but I'm gonna guess you had your reasons at the time. 
As it seems to work fine with the disable websecurity flag I'm not really keen on fixing what ain't broke.

Hoping we can merge this and get it on the main dash docset repo.

jariz committed:
The selector was fine to begin with - woops.
